artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -111,9 +111,13 @@ public static org.apache.activemq.artemis.api.core.Message inbound(final Message
       final ActiveMQBuffer body = coreMessage.getBodyBuffer();
 
       final ByteSequence contents = messageSend.getContent();
-      if (contents == null && coreType == org.apache.activemq.artemis.api.core.Message.TEXT_TYPE) {
-         body.writeNullableString(null);
-      } else if (contents != null) {
+      if (contents == null) {
+         if (coreType == org.apache.activemq.artemis.api.core.Message.TEXT_TYPE) {
+            body.writeNullableString(null);
+         } else if (coreType == org.apache.activemq.artemis.api.core.Message.MAP_TYPE) {
+            body.writeByte(DataConstants.NULL);
+         }
```

Comment:
The new tests dont appear to cover this case, the main addition of the PR and how the issue was found?

Also feels like this could be unit tested rather than system tested.

Reply:
Not all of the tests cover this specific issue. Most of them, in fact, cover other uses cases involving the same JMS code with different protocols (i.e. core and AMQP). However, I felt it was worth confirming those work as expected.

That said, both `testMapMessageConversionBetweenOpenWireAndAMQP` and `testMapMessageConversionBetweenOpenWireAndCore` fail without the fix.

I'll add a unit test for good measure.